sdk/search/azure-search/src/main/java/com/azure/search/documents/util/RegexFlagsUtils

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.util;
+
+import com.azure.search.documents.models.RegexFlags;
+import java.util.Arrays;
+
+/**
+ * Utility class for {@link RegexFlags} complex operations.
+ */
+public final class RegexFlagsUtils {
+
+    /**
+     * Combines a list of {@link RegexFlags} into one separated by "|".
+     * E.g. {@code "CANON_EQ|CASE_INSENSITIVE|COMMENTS"}
+     *
+     * @param regexFlags A list a {@link RegexFlags} to combine.
+     * @return New combined {@link RegexFlags} separated by vertical bar.
+     */
+    public static RegexFlags combineRegexFlags(RegexFlags... regexFlags) {
+        return RegexFlags.fromString(String.join("|", Arrays.deepToString(regexFlags)));
```

Comment:
This won't work as expected for a few reasons.

- If `regexFlags` is `null` this will return `"null"`.
- If `regexFlags` has values it will result in `"[val1.toString(), val2.toString(), ...]"`.

This would be a better option.

```java
if (CoreUtils.isNullOrEmpty(regexFlags)) {
    return null;
}

return Arrays.stream(regexFlags)
    .map(RegexFlags::toString)
    .collect(Collectors.joining("|"));
```